### PR TITLE
feat(replays): add replays recording segment model

### PR DESCRIPTION
**Pull request by JoshFerge**
- Adds a model, `ReplayRecordingSegment` which will link a recording segment (replay_id+sequence_id) to the `File` for which the blob is stored at.

For the next several months, this table should not receive a lot of volume, no more than 1-10 a second. we will add rate limits upstream to ensure this. Thinking further out, we may use a different solution for file blobs, depending on how well this can scale / other constraints.

See https://www.notion.so/sentry/Replay-Backend-Architecture-Update-7-19-7c4ecc8af3fb4027992f3ee75dae9529 for our current backend architectural overview for replays.

The consumer PR which will use this model is coming shortly.

### CODE REVIEW

File: src/sentry/replays/models.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    class Meta:
+        app_label = "replays"
+        db_table = "replay_recordingsegment"
+        index_together = (("replay_id", "sequence_id"),)
+        unique_together = (
+            ("project_id", "replay_id", "file_id"),
+            ("project_id", "replay_id", "sequence_id"),
+        )
+
+    __repr__ = sane_repr("event_id", "name", "file_id")
```
Comment (wedamija) [inline src/sentry/replays/models.py:32]: Need to update this

**Review by wedamija — COMMENTED** [review]
We should also add this to our cleanup script here: https://github.com/getsentry/sentry/blob/5476657d64de0916c319aa39e453bf8fb2c5132f/src/sentry/runner/commands/cleanup.py#L187-L189

Since we will want to put this on the `files` db in production we'll need a getsentry pr as well. @beezz do we have a process for doing this? Should we just put routings in place in getsentry beforehand so that the table is created there?

File: src/sentry/replays/migrations/0001_init_replays.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        migrations.CreateModel(
+            name="ReplayRecordingSegment",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("project_id", sentry.db.models.fields.bounded.BoundedBigIntegerField()),
```
Comment (markstory) [inline src/sentry/replays/migrations/0001_init_replays.py:42]: Now that you're in the same database as project do you want to have a foreign key?

You should also consider adding ReplayRecordingSegment to the [list of children for Project in the deletions task](https://github.com/getsentry/sentry/blob/8be757c0a7d0394e297cf099a9a1c8a3fc7edc65/src/sentry/deletions/defaults/project.py#L17).

Comment (JoshFerge) [reply src/sentry/replays/migrations/0001_init_replays.py:42]: added to the deletions task. if we decide to split this out into its own database, would project_id being a foreign key make that any more difficult?

Comment (wedamija) [reply src/sentry/replays/migrations/0001_init_replays.py:42]: Are we putting this in the same database as project? I thought we were going to put it on the `files` db?

Comment (JoshFerge) [reply src/sentry/replays/migrations/0001_init_replays.py:42]: I don't think it matters which database its on to start while we're in early alpha / beta, but yeah i think a separate database long term is the right move. i'm not sure if `files`, or `attachments` or a new database entirely would be the right home for it. in any case we can leave this as not-a-foreign-key

**Comment by beezz** [comment]
> We should also add this to our cleanup script here:
> 
> https://github.com/getsentry/sentry/blob/5476657d64de0916c319aa39e453bf8fb2c5132f/src/sentry/runner/commands/cleanup.py#L187-L189
> 
> Since we will want to put this on the `files` db in production we'll need a getsentry pr as well. @beezz do we have a process for doing this? Should we just put routings in place in getsentry beforehand so that the table is created there?

That's right, just update the db router before merging this so the migration is executed on the correct db. I guess I'd prefer we put this on `attachments` as it hosts a smaller dataset.

We also need to configure attachments/files to run migrations ~ https://github.com/getsentry/getsentry/blob/master/getsentry/conf/settings/prod.py#L50

**Comment by JoshFerge** [comment]
https://github.com/getsentry/getsentry/pull/7914 and https://github.com/getsentry/getsentry/pull/7915 merged, this should be good to go.

**Review by wedamija — APPROVED** [review]

File: src/sentry/replays/migrations/0001_init_replays.py (hunk 2 above)
Comment (markstory) [reply src/sentry/replays/migrations/0001_init_replays.py:42]: If you do put these models on the `files` db, you'll also need to update `getsentry upgrade` to run migrations for the `files` connection in `conf/settings/prod.py`. Currently migrations do not run on the `files` primaries during deploys.

**Pull request merged**